test_backprop.py does not pass in debug mode

Version: current master 03157f852002dcdcb04e03e6da03315976ce7434

```
$ CHAINER_DEBUG=1 pytest tests/chainer_tests/test_backprop.py
```

#### Result

```
E   AttributeError: 'NoneType' object has no attribute 'dtype'
 TestFunctionOutputNone_use_chainerx_false__chainerx_device_None__use_cuda_true__cuda_device_0__use_cudnn_never__cudnn_deterministic_false__autotune_false__cudnn_fast_batch_normalization_false__use_ideep_never_param_1_{function_node=False}.
test_backprop
tests/chainer_tests/test_backprop.py:483: in test_backprop
    y1, y2, y3 = self.call_func(backend_config, x1)
tests/chainer_tests/test_backprop.py:472: in call_func
    return self.call_func_function(backend_config, x1)
tests/chainer_tests/test_backprop.py:422: in call_func_function
    return Func()(x1)
chainer/function.py:269: in __call__
    ret = node.apply(inputs)
chainer/function_node.py:343: in apply
    for out in outputs):
chainer/function_node.py:343: in <genexpr>
    for out in outputs):
chainer/backend.py:31: in _contains_nan
    if x.dtype.kind in ('f', 'c'):
E   AttributeError: Parameterized test failed.
E
E   Base test method: TestFunctionOutputNone_use_chainerx_false__chainerx_device_None__use_cuda_true__cuda_device_0__use_cudnn_never__cudnn_deterministic_false__autotune_false__cudnn_fast_batch_normalization_false__use_ideep_never.test_back
prop
E   Test parameters:
E     function_node: False
E
E
E   (caused by)
E   AttributeError: 'NoneType' object has no attribute 'dtype'
```

--- a/chainer/function_node.py
+++ b/chainer/function_node.py
@@ -339,11 +339,11 @@ Use apply() method instead.\
 
         # NaN check of output values
         if is_debug:
-            if any(chainer.backend._contains_nan(out)
-                   for out in outputs):
-                msg = ('NaN is detected on forward computation of '
-                       '{}'.format(self.label))
-                raise RuntimeError(msg)
+            for out in outputs:
+                if out is not None and chainer.backend._contains_nan(out):
+                    msg = ('NaN is detected on forward computation of '
+                           '{}'.format(self.label))
+                    raise RuntimeError(msg)
 
         self._output_count = len(outputs)
 

